File: javascript/1047. Remove All Adjacent Duplicates In String.java
class Solution {
    public String removeDuplicates(String S) {
        Deque<Character> result = new ArrayDeque();
        StringBuffer letters =new StringBuffer();
        result.push(S.charAt(0));
        
        for(int index = 1; index < S.length(); index++){
            char character = S.charAt(index);
            if(result.size() > 0 && result.peekLast() == character){
                result.pollLast();
                continue;
            }
            result.offer(character);
        }
        
        for(char s:result){
            letters.append(s);
        }
        return letters.toString();
    }
}

File: javascript/1047. Remove All Adjacent Duplicates In String.js
var removeDuplicates = function(S) {
    let result = [S.charAt(0)];
    for(let index = 1; index < S.length; index++){
        let length = result.length;
        let char = S.charAt(index);
        if(result[length-1] == char){
            result.pop();
            continue;
        }
        result.push(char);
    }
    return result.join('');
};
